src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -387,6 +389,75 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
                 EmailType.FEEDBACK_UNPUBLISHED.getSubject());
     }
 
+    public EmailWrapper generateFeedbackSessionResendLinksEmail(String userEmail) {
+        Date endTime = new Date();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.MONTH, -6);
+        Date startTime = calendar.getTime();
+        String subject = EmailType.FEEDBACK_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String emailBody;
+        String studentName = "";
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllOpenFeedbackSessions(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
+            CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+            StudentAttributes student = studentsLogic.getStudentForEmail(course.getId(), userEmail);
+
+            if (student != null) {
+                studentName = student.getName();
+                String submitUrlHtml = "(Feedback session is not yet opened)";
+                String reportUrlHtml = "(Feedback session is not yet published)";
+
+                if (session.isOpened() || session.isClosed()) {
+                    String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    submitUrlHtml = "<a href=\"" + submitUrl + "\">" + submitUrl + "</a>";
+                }
+
+                if (session.isPublished()) {
+                    String reportUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_RESULTS_PAGE)
+                            .withCourseId(course.getId())
+                            .withSessionName(session.getFeedbackSessionName())
+                            .withRegistrationKey(StringHelper.encrypt(student.key))
+                            .withStudentEmail(student.email)
+                            .toAbsoluteString();
+                    reportUrlHtml = "<a href=\"" + reportUrl + "\">" + reportUrl + "</a>";
+                }
+
+                linksFragmentValue.append(Templates.populateTemplate(
+                        EmailTemplates.FRAGMENT_SINGLE_FEEDBACK_SESSION_LINKS,
+                        "${feedbackSessionName}", session.getFeedbackSessionName(),
+                        "${deadline}", session.getEndTimeString() + (session.isClosed() ? " (Passed)" : ""),
+                        "${submitUrl}", submitUrlHtml,
+                        "${reportUrl}", reportUrlHtml));
+            }
```

Comment:
I think we need `courseName` and `courseId` here as well, for the cases when student is in more than 1 course.